### PR TITLE
update-store-retargetting-data

### PR DESCRIPTION
**Pull request by IloveJavaa**
Stored the retargetting data.
Asana link - https://app.asana.com/0/0/1206458202841357/f

### CODE REVIEW

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Constants.kt [hunk 1]
```diff
@@ -54,6 +54,7 @@ object Constants {
     const val SHARED_PREF_LAST_SESSION_TIME = "last_session_time"
     const val SHARED_PREF_FIRST_INSTALL = "first_install"
     const val DEVICE_BRAND = "xiaomi"
+    const val STORE_RETARGETTING = "retargetting"
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Constants.kt:57]:
```suggestion
    const val STORE_RETARGETING = "retargeting"
```

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]
fix retargeting word

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt [hunk 2]
```diff
@@ -59,6 +59,13 @@ object Util {
 
         return printHexBinary(sha256Hmac.doFinal(data.toByteArray()))
     }
+    
+    fun retargettingData(url: String, time: String): String {
+        val body = mutableMapOf<String, Any>()
+        body["date&time"] = time
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:65]: key name should not have &

```suggestion
        body["rtgtime"] = time
```

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierWorkRequest.kt [hunk 3]
```diff
@@ -88,6 +89,7 @@ class TrackierWorkRequest(
         body["cphone"] = customerPhoneNumber
         body["cname"] = customerName
         body["getPreLoadAndPAIdata"] = preinstallData.toString()
+        body["storeRetargetting"] = Util.retargettingData(storeRetargetting, Util.dateFormatter.format(Date()))
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierWorkRequest.kt:92]: why are you sending the current time here?

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt [hunk 4]
```diff
@@ -220,4 +220,9 @@ object TrackierSDK {
         instance.dob = dob
     }
     
+    @JvmStatic
+    fun storeRetargetting(context: Context, uri: String) {
+        val ctx = context.applicationContext
+        Util.setSharedPrefString(ctx, Constants.STORE_RETARGETTING, uri)
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:226]: when storing the retargeting info here you have to store the time as well

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Constants.kt (hunk 1 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Constants.kt:57]: Fixed

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt (hunk 2 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:65]: fixed

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierWorkRequest.kt (hunk 3 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierWorkRequest.kt]: I have changed this from here now

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 5]
```diff
@@ -335,4 +338,11 @@ class TrackierSDKInstance {
         Util.setSharedPrefString(this.config.context, Constants.SHARED_PREF_DEEP_LINK_CALLED, "true")
         dlt.onDeepLinking(dlResult)
     }
-}
+    
+    fun getRetargetingData(): String {
+        val body = mutableMapOf<String, Any>()
+        body["rtgtime"] = Util.getSharedPrefString(this.config.context, Constants.STORE_RETARGETING_TIME)
+        body["url"] = Util.getSharedPrefString(this.config.context, Constants.STORE_RETARGETING)
+        return body.toString()
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:346]: return body as it is @IloveJavaa

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]
changes needed

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt (hunk 5 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:346]: done

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**